pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -1870,3 +1860,178 @@ func uplinkToPhysdev(ifname string) string {
 	log.Functionf("uplinkToPhysdev found %s", kernIfname)
 	return kernIfname
 }
+
+func isValidIPRange(ipRange types.IpRange) bool {
+	if ipRange.Start == nil || ipRange.End == nil ||
+		ipRange.Start.IsUnspecified() || ipRange.End.IsUnspecified() {
+		return false
+	}
+	return true
+}
+
+func getIPRangeAddrCount(ipRange types.IpRange) int {
+	var start, end uint
+	start = uint(ipRange.Start[0])<<24 + uint(ipRange.Start[1])<<16 +
+		uint(ipRange.Start[2])<<8 + uint(ipRange.Start[3])
+	end = uint(ipRange.End[0])<<24 + uint(ipRange.End[1])<<16 +
+		uint(ipRange.End[2])<<8 + uint(ipRange.End[3])
+	return int(end - start)
+}
+
+func isValidIPSubnet(subnet net.IPNet) bool {
+	if subnet.IP == nil || subnet.Mask == nil ||
+		subnet.IP.IsUnspecified() {
+		return false
+	}
+	if len(subnet.IP) == 0 || len(subnet.Mask) == 0 {
+		return false
+	}
+	setFlag := false
+	for i := 0; i < len(subnet.IP); i++ {
+		if subnet.IP[i] != 0 {
+			setFlag = true
+			break
+		}
+	}
+	if setFlag == false {
+		return false
+	}
+	setFlag = false
+	for i := 0; i < len(subnet.Mask); i++ {
+		if subnet.Mask[i] != 0 {
+			setFlag = true
+			break
+		}
+	}
+	if setFlag == false {
+		return false
+	}
+	netStr := subnet.String()
+	if netStr == "" {
+		return false
+	}
+	if _, _, err := net.ParseCIDR(netStr); err != nil {
+		return false
+	}
+	return true
+}
+
+func getIPSubnetAddrCount(subnet net.IPNet) int {
+	networkStr := subnet.String()
+	maskLenStr := strings.Split(networkStr, "/")[1]
+	maskLen, _ := strconv.Atoi(maskLenStr)
+	addrcount := 1 << (32 - maskLen)
+	// discount the network, gateway and broadcast addresses(3)
+	addrcount -= 3
+	return addrcount
+}
+
+// do a basic validation for the config
+func isValidIPConfig(ctx *zedrouterContext,
```

Comment:
N layers of tiny functions makes the code harder to understand. This one isn't needed.
The one caller can just check if !isValidIPRange() && !isValidIPSubnet().

But even easier to understand would be to have the one caller just ask for the counts e.g., only keep 
getIPRangeAddrCount() and getIPSubnetAddrCount() and have them both return (int, error). 
That why you can drop all of the *Valid* functions and booleans.

Reply:
Simplified, as per the suggestion